File: rb/features.rb
# ShortLink SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ShortLinkFeatures
  def self.make_feature(name)
    case name
    when "base"
      ShortLinkBaseFeature.new
    when "ratelimit"
      ShortLinkRatelimitFeature.new
    when "retry"
      ShortLinkRetryFeature.new
    when "test"
      ShortLinkTestFeature.new
    when "timeout"
      ShortLinkTimeoutFeature.new
    else
      ShortLinkBaseFeature.new
    end
  end
end

File: py/features.py
# ShortLink SDK feature factory

from feature.base_feature import ShortLinkBaseFeature
from feature.test_feature import ShortLinkTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ShortLinkBaseFeature(),
        "test": lambda: ShortLinkTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
